src/Subject.ts

```diff
@@ -29,7 +29,7 @@ export class Subject<T> extends Observable<T> implements Observer<T>, Subscripti
   protected dispatching: boolean = false;
   protected hasCompleted: boolean = false;
 
-  lift<T, R>(operator: Operator<T, R>): Observable<T> {
+  lift<R>(operator: Operator<T, R>): Observable<T> {
```

Comment:
@david-driscoll Ok, I've committed my partial fix for this. From my understanding the signature here should look like:

```
  lift<R>(operator: Operator<T, R>): Observable<R> {
    const subject = new Subject<R>(this, this.destination || this);
    subject.operator = operator;
    return subject;
  }
```

The problem now is that the first line of the function tries to produce a `Subject<R>` using `this`, which doesn't make sense. I'm not sure what the fix is.